packages/mirrorful/editor/src/pages/index.tsx

```diff
@@ -1,8 +1,38 @@
 import { Dashboard } from '@mirrorful/core/lib/components/Dashboard'
+import ServerEndedMessage from '@mirrorful/core/lib/components/ServerEndedMessage'
 import { TConfig } from '@mirrorful/core/lib/types'
 import Head from 'next/head'
+import { useState, useRef, useEffect } from 'react'
 
 export default function Editor() {
+  const [hasShutDown, setHasShutDown] = useState(false)
+  const counter = useRef<boolean>(false)
```

Comment:
What's the reason for the `counter`? It also doesn't make a ton of sense to me that it's a Ref? I feel like it should just be state? I'm curious though!

Reply:
Yeah, the counter needs to be changed to a different name. But essentially in testing useState did not work by itself it needed a ref. I can try again soon to see if it works now!